### PR TITLE
Closes #352 Updated method in abstract dto

### PR DESCRIPTION
**Pull request by OlehPaliuh**

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/controller/dto/annotations/PriceEntityInterface.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.softserve.academy.spaced.repetition.controller.dto.annotations;
+
+import com.softserve.academy.spaced.repetition.domain.BasePrice;
+
+public interface PriceEntityInterface {
+
+    BasePrice getPriceObject();
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/dto/annotations/PriceEntityInterface.java]: This method:

- must be removed in the interface, as it is not accessed outside of implementation.

- can be removed in the implementation.

File: src/main/java/com/softserve/academy/spaced/repetition/domain/BasePrice.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.softserve.academy.spaced.repetition.domain;
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+
+@MappedSuperclass
+public class BasePrice {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/BasePrice.java]: Please make it abstract to guarantee that no instance of this class can be created.

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Course.java [hunk 3]
```diff
@@ -203,4 +204,20 @@ public int hashCode() {
         result = 31 * result + (coursePrice != null ? coursePrice.hashCode() : 0);
         return result;
     }
+
+    @Override
+    public BasePrice getPriceObject() {
```
[the remark this answers is not in the document]
Comment (antonbakalets) [reply src/main/java/com/softserve/academy/spaced/repetition/domain/Course.java]: See my comment on PriceEntityInterface

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1.13__Update_deckPrice_column_name.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+ALTER TABLE deck_price CHANGE COLUMN deck_price price INT NOT NULL default 0;
```
Comment (antonbakalets) [inline src/main/resources/db/migration/V1.13__Update_deckPrice_column_name.sql:1]: Just a double check: don't you need the same for COURSE_PRICE?

**Review by antonbakalets — APPROVED** [review]

File: src/main/resources/db/migration/V1.13__Update_deckPrice_column_name.sql (hunk 4 above)
Comment (OlehPaliuh) [reply src/main/resources/db/migration/V1.13__Update_deckPrice_column_name.sql:1]: No, because in the course_price table column named price.

**Comment by antonbakalets** [comment]
What is preventing the merge?

**Review by volodymyrglynka — APPROVED** [review]

**Review by Maks-Chornyi — APPROVED** [review]

**Pull request merged**